test/checks/navigation/identical-links-same-purpose.js

```diff
@@ -0,0 +1,266 @@
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when ARIA link that has no accessible name', function() {
+		var vNode = queryFixture('<span role="link" id="target"></span>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when native link has only emoji as accessible name', function() {
+		var vNode = queryFixture(
+			'<a id="target" href="/some-directory/contact/rock.html">🤘</a>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when native link has only nonBmp characters (diacritical marks supplement) as accessible name', function() {
+		var vNode = queryFixture(
+			'<a id="target" href="/some-directory/contact/rock.html">ᴁ</a>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when native link has only nonBmp characters (currency symbol) as accessible name', function() {
+		var vNode = queryFixture(
+			'<a id="target" href="/some-directory/currency.html">₨ </a>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when ARIA link has only punctuations as accessible name', function() {
+		var vNode = queryFixture('<button id="target" role="link">!!!!</button>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns undefined when ARIA link has only combination of emoji, punctuations, nonBmp characters as accessible name', function() {
+		var vNode = queryFixture(
+			'<button id="target" role="link">☀️!₨   </button>'
+		);
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isUndefined(actual);
+		assert.isNull(checkContext._data);
+	});
+
+	it('returns true for native links with `href` and accessible name', function() {
+		var vNode = queryFixture('<a id="target" href="/home/#/foo">Pass 1</a>');
+		var actual = check.evaluate.call(
+			checkContext,
+			vNode.actualNode,
+			options,
+			vNode
+		);
+		assert.isTrue(actual);
+		assert.hasAllKeys(checkContext._data, ['name', 'parsedResource']);
+		assert.equal(checkContext._data.name, 'Pass 1'.toLowerCase());
+		assert.equal(checkContext._data.parsedResource.hash, '#/foo');
```

Comment:
Simpler way to test this would be just to test that whatever is on `.parsedResource` is deepEqual to what you get from `dom.parsedResource`.